File: src/main/java/org/apache/prepbuddy/datacleansers/imputation/ModeSubstitution.java
package org.apache.prepbuddy.datacleansers.imputation;

import org.apache.prepbuddy.groupingops.TextFacets;
import org.apache.prepbuddy.rdds.TransformableRDD;
import org.apache.prepbuddy.utils.RowRecord;
import scala.Tuple2;

import java.util.List;

/**
 * This imputation strategy impute the column value by the mode of that column
 */
public class ModeSubstitution implements ImputationStrategy {
    private  Tuple2 mode;
    @Override
    public void prepareSubstitute(TransformableRDD rdd, int missingDataColumn) {
        TextFacets textFacets = rdd.listFacets(missingDataColumn);
        List<Tuple2> listOfHighest = textFacets.highest();
        mode = listOfHighest.get(0);
    }

    @Override
    public String handleMissingData(RowRecord record) {
        return mode._1().toString();
    }
}

File: src/main/scala/org/apache/datacommons/prepbuddy/cleansers/imputation/ModeSubstitution.scala
package org.apache.datacommons.prepbuddy.cleansers.imputation

import org.apache.datacommons.prepbuddy.cluster.TextFacets
import org.apache.datacommons.prepbuddy.rdds.TransformableRDD
import org.apache.datacommons.prepbuddy.utils.RowRecord

case class ModeSubstitution() extends ImputationStrategy{
    private val mode: String = null
    override def prepareSubstitute(rdd: TransformableRDD, missingDataColumn: Int): Unit = {
        val listFacets: TextFacets = rdd.listFacets(missingDataColumn)
    }

    override def handleMissingData(record: RowRecord): String = ???
}
